components/tooltip/index.tsx

```diff
@@ -71,7 +70,7 @@ interface LegacyTooltipProps
 export interface AbstractTooltipProps extends LegacyTooltipProps {
   style?: React.CSSProperties;
   className?: string;
-  color?: LiteralUnion<PresetColorType, string>;
+  color?: PresetColorType | React.CSSProperties['color'];
```

Comment:
这样写 TS 类型提示正常么

Reply:
```suggestion
  color?: PresetColorType | string & {};
```
感觉是不是这样定义类型就好了，同时支持字面量和扩展字符串

Reply:
之前这里讨论过这个问题 https://github.com/ant-design/ant-design/pull/39394/files#r1043160820 我这边用的 VSCode不清楚其他编辑器是什么结果。 另外我可以用一个视频来补充一下为什么这样改

Reply:
创建了一个 [Typescript Play](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgIilQ3wChSkAPSWONCAOwGd4wimkYBhCAG2iZwAvHADapOHADkRACZSANBOkBzIkgaLlUgEa8Arki2SpATyS9+Ady0BdOCkH1mMcjFNgkcAPoAFdpw8-FCCIu6eEJhwbEgc3HwCogz6IDpIULak4V4AkgwAbukwSLL+sYEJIcJwAAYAJADefgHxwUwAvgC0wAVFJTVuHl5lcUEC1c3lreMAPnB5hbAlIxVt5D3FUJjoXmNQXAAW+gwA1v4QYIINyvTBAFxwK9NVc8joMAB0XADK3+eesGAsVEUlu0Ck9gA9JC4DADsBBFZoCcmMpoXRKg8nntBHMWFAeio4AAyOANdpwdFwhFwJFQFE3eG8WREBgAfgebwwHy5MAAchBZEhSO1yBRqNB4M4WHA9odjidqgAKBoY4IKOhMlkaODtB5yo6nf5MACUwgAfMoADxMMAoBhwFimXhIIQNVVgqC69rmhpoLWs9pWyG2+2W8jS+AAQTAYGVZqE5rgSutlskkitBoVaugQkIJXw5uQsmDWdOafTmcq8tOOagebk3V6S1kheLnQWfRLkLLJwrkgakIAVHAGBBWC04EPIaL03Aq8Ea4rPXmrPDioW18BiqXq4a+8oM726w2VDoUEqAAwa69wW8fABMJrbZ5Qu8X+4rwctJvIVBo8BCts+i8NGsakEAA)， @kiner-tang 的建议会更好一些， 但是之前的 LiteralUnion 在我的编辑器就不生效，我需要检查一下

Reply:
LiteralUnion 就是这么实现的

Reply:
可以把原来的 LiteralUnion 直接修改为 `type LiteralUnion<T extends string> = T | (string & {})`